quiche/src/lib.rs

```diff
@@ -7984,6 +7985,95 @@ impl std::fmt::Debug for Stats {
     }
 }
 
+/// QUIC Unknown Transport Parameter
+///
+/// A QUIC transport parameter that is not specifically recognized
+/// by this implementation.
+#[derive(Clone, Debug, PartialEq)]
+pub struct UnknownTransportParameter {
+    /// ID of an unknown transport parameter
+    pub id: u64,
+    /// Original data representing the value of an unknown transport parameter
+    pub value: Vec<u8>,
+}
+
+#[cfg(feature = "qlog")]
+impl From<UnknownTransportParameter>
+    for qlog::events::quic::UnknownTransportParameter
+{
+    fn from(value: UnknownTransportParameter) -> Self {
+        Self {
+            id: value.id,
+            value: value.value,
+        }
+    }
+}
+
+/// Track unknown transport parameters, up to a limit.
+#[derive(Clone, Debug, PartialEq)]
+pub struct UnknownTransportParameters {
+    // The maximum number of bytes available to store the values of unknown
+    // transport parameters.
+    /// The space remaining for storing unknown transport parameters.
+    pub capacity: usize,
+    /// The unknown transport parameters.
+    pub parameters: Vec<UnknownTransportParameter>,
+}
+
+impl<'a> Default for UnknownTransportParameters {
+    fn default() -> Self {
+        Self {
+            capacity: Self::MAX_UNKNOWN_TRANSPORT_PARAM_SIZE,
+            parameters: vec![],
+        }
+    }
+}
+
+impl UnknownTransportParameters {
+    const MAX_UNKNOWN_TRANSPORT_PARAM_SIZE: usize = 256;
+
+    /// Push an unknown transport parameter into storage if there is space
+    /// remaining.
+    pub fn safe_push(&mut self, new: UnknownTransportParameter) -> Result<()> {
+        let new_unknown_tp_size = new.value.len();
```

Comment:
If I read this correct. it might allow an "infinite" amount of zero-length TPs. So the capacity limit should probably factor in the size of the TP identifier two (8 bytes). Probably easiest to add a `len()` function to UnknownTransportParameter type.

Reply:
You are absolutely correct -- good eyes! I have fixed that in the most recent version!